en/guide/extending.md

```diff
@@ -15,39 +15,35 @@ Video Streaming | Developers can implement new types of video streaming (e.g. HL
+Developers may want to add support for a new type of camera device that is not currently supported by DCM. To support a custom type of camera, a `PluginXXX` class derived from `PluginBase` and `CameraDeviceXXX` class derived from `CameraDevice` must be implemented. An example of a plugin for a custom camera device with classes `PluginCustom`(https://github.com/Dronecode/camera-manger/tree/master/plugins) and `CameraDeviceCustom`(https://github.com/Dronecode/camera-manger/tree/master/plugins) is added in the project and may be referenced.
+The sections below detail these steps.
 
-### 1. Extend CameraDevice Class
+### 1. Extend PluginBase Class
 
-To add support for new type of camera device in DCM, a custom class must be derived from `CameraDevice`. 
-An example `CameraDeviceCustom` (in green) can be seen in the [class diagram](../guide/architecture.md#class_diagram). 
-The class `CameraDeviceCustom` represents a custom type of camera device.
+Plugins are self-registering objects that discovers, enlists and creates camera devices.
+
+**Action**: `PluginCustom` must implement the pure virtual functions of the base class `PluginBase`.
+
+The plugin will prepare a list of camera devices that are attached to the system.
+
+**Action**: `PluginCustom` must implement a logic to discover camera devices.
+
+### 2. Extend CameraDevice Class
+
+To add support for new type of camera device in DCM, `CameraDeviceCustom` class must be derived from `CameraDevice`. 
 
 **Action**: `CameraDeviceCustom` must implement the pure virtual functions of the base class `CameraDevice`. 
 
-```cpp
-class CameraDeviceCustom final : public CameraDevice {
-public:
-    CameraDeviceGazebo(std::string device);
-    ~CameraDeviceGazebo();
-    std::string getDeviceId();
-    int getInfo(struct CameraInfo &camInfo);
-    bool isGstV4l2Src();
-    int init(CameraParameters &camParam);
-    int uninit();
-    int resetParams(CameraParameters &camParam);
-    int setParam(CameraParameters &camParam, std::string param, const char *param_value,
-                 size_t value_size, int param_type);
-```
+There are few methods in `CameraDevice` class that are not pure virtual. By default it returns `NOT_SUPPORTED`.
 
-**Action**: `CameraDeviceCustom` may overload other methods to provide the functionality.
+**Action**: `CameraDeviceCustom` may overload not-pure virtual methods to provide the functionality.
 
 The configurable parameters of the custom camera device can be exported to the client (GCS) for control. Setting of these parameters and resetting of all the parameters must be handled by the `CameraDeviceCustom` class.
 
 **Action*: `CameraDeviceCustom` must declare the parameter name (string), ID (int) and type (enum). Also set the default value of the parameter.
```

Comment:
Again, not sure that we really need the "Action" headings. But if you want to keep them, this one has a missing asterisk.

Reply:
So maybe something like 

> `CameraDeviceCustom` may declare the parameters that are relevant to the camera type. Each parameter has a   name (string), ID (int), type (enum) and default value.

What about the case where a camera can have unique parameters? ie it implements some generic API, but might have additional parameters that aren't part of the API itself (ie that might be set by camera config file).
I guess what I'm saying is what parameters get set here, and which get set in a camera definition file?